### PR TITLE
Use cursor-based pagination for build events

### PR DESCRIPTION
**Pull request by aoldershaw**
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## What does this PR accomplish?

Bug Fix | **Feature** | Documentation

...rather than LIMIT+OFFSET, which doesn't scale well with many build
events (see https://ivopereira.net/efficient-pagination-dont-use-offset-limit)

for instance, build 1139906 on CI has 600000 build events - fetching a
single batch with offset 500000 takes over 1s using LIMIT+OFFSET, while
only takes 10ms using the cursor-based approach (event_id > $2)

also, rather than incrementing the cursor for each row we scan, it feels
safer to set the cursor to the last event id - I'm not positive it'll
help, but I'm hoping it might prevent issues like #6828 - without a good
repro, though, it's hard to know

## Release Note

* Optimizes fetching build logs from the DB for builds with massive logs

## Contributor Checklist

- [ ] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [ ] [Signed] all commits
- [ ] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Review by chenbh — APPROVED** [review]
lgtm

Did we ever come to a conclusion on if we want to build sql via raw sql statements (`tx.Query('SELECT...')`) or via functions (`psql.Select().From().Where()`)?

**Comment by aoldershaw** [comment]
@chenbh not sure in general - I think `sq` makes some queries nicer, but sometimes feels constraining (in which case raw SQL is better IMO). Don't think the queries here fall into that camp, though, so converted them to the `sq` form in https://github.com/concourse/concourse/pull/6873/commits/be11381ef36116d9f4289a3551774077514a75df

File: atc/db/build_event_source.go [hunk 1]
```diff
@@ -108,25 +109,28 @@ func (source *buildEventSource) collectEvents(from uint) {
...
+		rows, err := psql.Select("event_id", "type", "version", "payload").
+			From(source.table).
+			Where(sq.Or{
+				sq.Eq{"build_id": source.buildID},
+				sq.Eq{"build_id_old": source.buildID},
+			}).
+			Where(sq.Gt{"event_id": cursor}).
+			Limit(uint64(batchSize)).
+			RunWith(tx).
+			Query()
```
Comment (chenbh) [inline atc/db/build_event_source.go]: Is this missing a `.OrderedBy(....)`?

Comment (aoldershaw) [reply atc/db/build_event_source.go]: yikes, good catch. fixed

**Review by chenbh — APPROVED** [review]
lgtm

**Pull request merged**